src/app/database/services/database.service/LRUCache.ts

```diff
@@ -84,9 +121,46 @@ export class LRUCache<T> {
       this.lruList.splice(index, 1);
     }
     this.lruList.push(key);
+    this.persistState();
+  }
+
+  private persistState(): void {
     this.cache.updateLRUState({
       lruKeyList: this.lruList.slice(),
       itemsListSize: this.currentSize,
     });
   }
+
+  private async reconcileState(): Promise<void> {
+    if (!this.lruList.length && this.currentSize === 0) {
+      this.persistState();
+      return;
+    }
+
+    const results = await Promise.all(
+      this.lruList.map(async (cachedKey) => {
+        const exists = await this.cache.has(cachedKey);
+        if (!exists) {
+          return null;
+        }
+        const cachedSize = await this.cache.getSize(cachedKey);
+        return { key: cachedKey, size: cachedSize };
```

Comment:
Same here, maybe we could reduce the complexiti extrating the logic to a simple method that makes one thing :)